### PR TITLE
feat(web-cloud.exchange/emailpro): add DKIM Autoconfig for both

### PR DESCRIPTION
**Pull request by antonymarion**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/dkim-b2` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat #MANAGER-12343 Feat #MANAGER-12208 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)

## Description

Add auto DKIM for emailPro and Exchange service.

### CODE REVIEW

File: packages/manager/modules/emailpro/src/dashboard/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -686,9 +686,14 @@
   "emailpro_tab_domain_diagnostic_spf_toolbox_ok": "Configuration automatique effectuée",
   "emailpro_tab_domain_diagnostic_dkim": "DKIM",
   "emailpro_tab_domain_diagnostic_dkim_title_error": "Diagnostic DKIM",
+  "emailpro_tab_domain_diagnostic_dkim_id_generation_no_ovhcloud": "La génération de vos identifiants sont en cours, veuillez patienter...",
```
Comment (JacquesLarique) [inline packages/manager/modules/emailpro/src/dashboard/translations/Messages_fr_FR.json:689]: I think there is a mistake, in my opinion it's the "génération" that is "en cours" and it should be "est en cours".

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
+export default class DkimAutoConfigurator {
+  initializeScope($scope) {
+    $scope.loadDataForDkim = () => this.loadDataForDkim();
+    $scope.getDkimSelector = () => this.getDkimSelector();
+    $scope.leaveDkimConfigurator = () => this.leaveDkimConfigurator();
+    $scope.getNextButtonDependingOnStep = () =>
+      this.getNextButtonDependingOnStep();
+    $scope.getTitleDependingOnStep = () => this.getTitleDependingOnStep();
+    $scope.initSpfContext = () => this.initSpfContext();
```
Comment (JacquesLarique) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js:9]: Is it really necessary to add these functions to the scope ?
Can't you access it through the controller where it's needed ?

File: packages/manager/modules/exchange/src/dashboard/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -413,12 +413,23 @@
...
   "exchange_tab_domain_diagnostic_dkim_configuration_nok": "Votre configuration est incorrecte, veuillez suivre les indications du <a href=\"{{url}}\" target=\"_blank\">guide</a>.",
-  "exchange_tab_domain_diagnostic_dkim_activation_no_ovhcloud": "Votre domaine n'étant pas hébergé chez OVHCloud nous ne pouvons pas le configurer automatiquement. Vous pouvez configurer DKIM en suivant ce <a href=\"{{url}}\" target=\"_blank\">guide</a>.",
-  "exchange_tab_domain_diagnostic_dkim_deactivation": "DKIM est actuellement actif sur votre domaine. Souhaitez-vous le desactiver ? (non recommandé)",
+  "exchange_tab_domain_diagnostic_dkim_activation_no_ovhcloud": "Votre domaine n'étant pas hébergé chez OVHCloud nous ne pouvons pas le configurer automatiquement. Souhaitez-vous débuter la configuration ?",
+  "exchange_tab_domain_diagnostic_dkim_configurate_no_ovhcloud": "Configurer",
+  "exchange_tab_domain_diagnostic_dkim_next_no_ovhcloud": "Suivant",
+  "exchange_tab_domain_diagnostic_dkim_confirm_no_ovhcloud": "Fermer",
+  "exchange_tab_domain_diagnostic_dkim_id_generation_no_ovhcloud": "La génération de vos identifiants sont en cours, veuillez patienter...",
```
Comment (JacquesLarique) [inline packages/manager/modules/exchange/src/dashboard/translations/Messages_fr_FR.json:422]: I think there is a mistake, in my opinion it's the "génération" that is "en cours" and it should be "est en cours".

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/dkim-auto-configurator.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
+export default class DkimAutoConfigurator {
+  initializeScope($scope) {
+    $scope.loadDataForDkim = () => this.loadDataForDkim();
+    $scope.getDkimSelector = () => this.getDkimSelector();
+    $scope.leaveDkimConfigurator = () => this.leaveDkimConfigurator();
+    $scope.getNextButtonDependingOnStep = () =>
+      this.getNextButtonDependingOnStep();
+    $scope.getTitleDependingOnStep = () => this.getTitleDependingOnStep();
+    $scope.initSpfContext = () => this.initSpfContext();
```
Comment (JacquesLarique) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/dkim-auto-configurator.js:9]: Is it really necessary to add these functions to the scope ?
Can't you access it through the controller where it's needed ?

**Review by JacquesLarique — CHANGES_REQUESTED** [review]

File: packages/manager/modules/emailpro/src/dashboard/translations/Messages_fr_FR.json (hunk 1 above)
Comment (antonymarion) [reply packages/manager/modules/emailpro/src/dashboard/translations/Messages_fr_FR.json:689]: Oups, Sorry I copy/pasted the french text given without checking it!

File: packages/manager/modules/exchange/src/dashboard/translations/Messages_fr_FR.json (hunk 3 above)
Comment (antonymarion) [reply packages/manager/modules/exchange/src/dashboard/translations/Messages_fr_FR.json:422]: Same as emailPro, I will change it to "est en cours" ;)

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js (hunk 2 above)
Comment (antonymarion) [reply packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js:9]: Yes of course, but I have chosen to keep the same logic as used previously (for configDkim for instance)

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/dkim-auto-configurator.js (hunk 4 above)
Comment (antonymarion) [reply packages/manager/modules/exchange/src/domain/dkim-autoconfig/dkim-auto-configurator.js:9]: Yes of course, but I have chosen to keep the same logic as used previously (for configDkim for instance)

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.html [hunk 5]
```diff
@@ -43,4 +37,78 @@
...
+                    <dd>
+                        <span data-ng-bind="ctrl.selector1RecordInfos"></span>
+                    </dd>
+                    <dd>
+                        <span data-ng-bind="ctrl.selector2NameInfos"></span>
+                    </dd>
+                    <dd>
+                        <span data-ng-bind="ctrl.selector2RecordInfos"></span>
+                    </dd>
+                </dl>
```
Comment (anooparveti) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.html:102]: **nitpick:** Use css utilities provided by ui-kit. 
Refer https://ovh.github.io/ovh-ui-kit/?path=/docs/design-system-content-descriptions--horizontal

```suggestion
                <dl class="oui-description oui-description_horizontal">
                    <dt
                        data-translate="exchange_tab_domain_diagnostic_dkim_text_field"
                    ></dt>
                    <dd>
                        <span data-ng-bind="ctrl.selector1NameInfos"></span>
                    </dd>
                    <dd>
                        <span data-ng-bind="ctrl.selector1RecordInfos"></span>
                    </dd>
                    <dd>
                        <span data-ng-bind="ctrl.selector2NameInfos"></span>
                    </dd>
                    <dd>
                        <span data-ng-bind="ctrl.selector2RecordInfos"></span>
                    </dd>
                </dl>
```

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.html [hunk 6]
```diff
@@ -34,4 +35,78 @@
...
+                    <dd>
+                        <span data-ng-bind="ctrl.selector1RecordInfos"></span>
+                    </dd>
+                    <dd>
+                        <span data-ng-bind="ctrl.selector2NameInfos"></span>
+                    </dd>
+                    <dd>
+                        <span data-ng-bind="ctrl.selector2RecordInfos"></span>
+                    </dd>
+                </dl>
```
Comment (anooparveti) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.html:100]: Refer https://ovh.github.io/ovh-ui-kit/?path=/docs/design-system-content-descriptions--horizontal
```suggestion
                <dl class="oui-description oui-description_horizontal">
                    <dt
                        data-translate="emailpro_tab_domain_diagnostic_dkim_text_field"
                    ></dt>
                    <dd>
                        <span data-ng-bind="ctrl.selector1NameInfos"></span>
                    </dd>
                    <dd>
                        <span data-ng-bind="ctrl.selector1RecordInfos"></span>
                    </dd>
                    <dd>
                        <span data-ng-bind="ctrl.selector2NameInfos"></span>
                    </dd>
                    <dd>
                        <span data-ng-bind="ctrl.selector2RecordInfos"></span>
                    </dd>
                </dl>
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.html [hunk 7]
```diff
@@ -43,4 +44,81 @@
...
+            data-wizard-on-finish="ctrl.leaveDkimConfigurator.bind(ctrl)"
+            data-wizard-on-cancel="ctrl.leaveDkimConfigurator.bind(ctrl)"
+            data-wizard-hide-previous-button="true"
+            data-wizard-confirm-button-text=":: 'exchange_tab_domain_diagnostic_dkim_confirm_no_ovhcloud' | translate"
+            data-wizard-next-button-text="ctrl.getNextButtonDependingOnStep() | translate"
+            data-wizard-title="ctrl.getTitleDependingOnStep() | translate"
+        >
+            <div
+                data-wizard-step
+                data-wizard-step-on-load="ctrl.getDkimSelector"
```
Comment (JacquesLarique) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.html:61]:
```suggestion
                data-wizard-step-on-load="ctrl.getDkimSelector.bind(ctrl)"
```
Is this not required ?

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.html [hunk 8]
```diff
@@ -34,4 +35,81 @@
...
+            data-wizard-on-finish="ctrl.leaveDkimConfigurator.bind(ctrl)"
+            data-wizard-on-cancel="ctrl.leaveDkimConfigurator.bind(ctrl)"
+            data-wizard-hide-previous-button="true"
+            data-wizard-confirm-button-text=":: 'emailpro_tab_domain_diagnostic_dkim_confirm_no_ovhcloud' | translate"
+            data-wizard-next-button-text="ctrl.getNextButtonDependingOnStep() | translate"
+            data-wizard-title="ctrl.getTitleDependingOnStep() | translate"
+        >
+            <div
+                data-wizard-step
+                data-wizard-step-on-load="ctrl.getDkimSelector"
```
Comment (JacquesLarique) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.html:52]:
```suggestion
                data-wizard-step-on-load="ctrl.getDkimSelector.bind(ctrl)"
```
Is this not required ?

Comment (JacquesLarique) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.html:52]:
```suggestion
                data-wizard-step-on-load="ctrl.getDkimSelector.bind(ctrl)"
```
Is this not required ?

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js [hunk 9]
```diff
@@ -1,23 +1,26 @@
 import { DKIM_CONFIGURATION_GUIDE } from './emailpro-domain-dkim-autoconfig.constants';
 
-export default class EmailProDomainDkimAutoconfigCtrl {
+import DkimAutoConfigurator from './dkim-auto-configurator';
+
+export default class EmailProDomainDkimAutoconfigCtrl extends DkimAutoConfigurator {
```
Comment (darsene) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:5]: Why do you split your controller in 2 classes whereas you don't use it anywhere else ?

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js [hunk 10]
```diff
@@ -1,6 +1,8 @@
 import { DKIM_CONFIGURATION_GUIDE } from './domain-dkim-autoconfig.constants';
 
-export default class ExchangeDomainDkimAutoconfigCtrl {
+import DkimAutoConfigurator from './dkim-auto-configurator';
+
+export default class ExchangeDomainDkimAutoconfigCtrl extends DkimAutoConfigurator {
```
Comment (darsene) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:5]: Why do you split your controller in 2 classes whereas you don't use it anywhere else ?

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js (hunk 9 above)
Comment (antonymarion) [reply packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:5]: I wanted to extract the logical part of dkim autoconfiguration which is the same between emailpro and exchange.

I kept it in each corresponding module

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js (hunk 10 above)
Comment (antonymarion) [reply packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:5]: Same

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    this.showConfiguratingBtn = true;
+    this.showSpfDiagnosticTitle = false;
+  }
+
+  async getSelectorNameForNoOvhCloud() {
+    if (!this.dkimSelectorsNoDomain) {
+      return;
+    }
+    const promises = this.dkimSelectorsNoDomain.map((dkimSelector) => {
+      return this.services.ExchangeDomains.getDkimSelectorName(
```
Comment (darsene) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js:20]:
```suggestion
      return this.services.EmailProDomains.getDkimSelectorName(
```
Normally, it shouldn't work

**Review by darsene — CHANGES_REQUESTED** [review]

**Review by darsene — APPROVED** [review]

**Review by JacquesLarique — APPROVED** [review]

**Pull request merged**